src/kernel/scheduler.zig

```diff
@@ -138,14 +138,12 @@ pub fn init(allocator: Allocator, mem_profile: *const mem.MemProfile) Allocator.
     tasks = TailQueue(*Task){};
 
     // Set up the init task to continue execution
-    current_task = try allocator.create(Task);
+    current_task = try Task.create(0, true, &vmm.kernel_vmm, allocator);
     errdefer allocator.destroy(current_task);
-    // PID 0
-    current_task.pid = 0;
+
+    allocator.free(current_task.kernel_stack);
```

Comment:
Might be worth adding a comment for why this free is here as a new kernel stack is allocated in `create`. Or have a way of not allocating a kernel stack in `Task.create`. Could have a `createWithStack` and you pass in address range for the kernel stack and a `createWithoutStack` which allocates the kernel stack. The naming of the functions can something else if you like. As allocating then immediately freeing memory does seem to be a bit dumb.

Reply:
Yeah I definitely agree. I'll add a function that creates with a stack.